File: app/src/test/java/com/mgaetan89/showsrage/fragment/PostProcessingFragment_CheckableToIntegerTest.java
package com.mgaetan89.showsrage.fragment;

import android.widget.Checkable;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

@RunWith(Parameterized.class)
public class PostProcessingFragment_CheckableToIntegerTest {
	@Parameterized.Parameter(0)
	public Checkable checkable;

	@Parameterized.Parameter(1)
	public boolean checked;

	@Parameterized.Parameter(2)
	public int result;

	@Before
	public void before() {
		if (this.checkable != null) {
			this.checkable.setChecked(this.checked);

			when(this.checkable.isChecked()).thenReturn(this.checked);
		}
	}

	@Test
	public void checkableToInteger() {
		assertThat(PostProcessingFragment.Companion.checkableToInteger(this.checkable)).isEqualTo(this.result);
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		return Arrays.asList(new Object[][]{
				{null, false, 0},
				{null, true, 0},
				{mock(Checkable.class), false, 0},
				{mock(Checkable.class), true, 1},
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/fragment/PostProcessingFragment_CheckableToIntegerTest.kt
package com.mgaetan89.showsrage.fragment

import android.widget.Checkable
import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized
import org.mockito.Mockito.`when`
import org.mockito.Mockito.mock

@RunWith(Parameterized::class)
class PostProcessingFragment_CheckableToIntegerTest(val checkable: Checkable?, val checked: Boolean, val result: Int) {
	@Before
	fun before() {
		if (this.checkable != null) {
			this.checkable.isChecked = this.checked

			`when`(this.checkable.isChecked).thenReturn(this.checked)
		}
	}

	@Test
	fun checkableToInteger() {
		assertThat(PostProcessingFragment.checkableToInteger(this.checkable)).isEqualTo(this.result)
	}

	companion object {
		@JvmStatic
		@Parameterized.Parameters
		fun data(): Collection<Array<Any?>> {
			return listOf(
					arrayOf(null, false, 0),
					arrayOf(null, true, 0),
					arrayOf<Any?>(mock(Checkable::class.java), false, 0),
					arrayOf<Any?>(mock(Checkable::class.java), true, 1)
			)
		}
	}
}
